tools/psoxy-test/lib/aws.js

```diff
@@ -75,7 +89,230 @@ async function call(options = {}) {
+    region: region,
+  }
+  if (role) {
+    let credentials;
+    try {
+      credentials = assumeRole(role);
+    } catch (error) {
+      throw new Error(`Unable to assume ${role}`, { cause: error });
+    }
+
+    options.credentials = {
+      // AWS CLI command will return credentials with 1st letter upper-case.
+      // However, S3 client expects different capitalization
+      ...Object.keys(credentials).reduce((memo, key) => {
+        memo[key.charAt(0).toLowerCase() + key.slice(1)] = credentials[key];
+        return memo;
+      }, {}),
+    }
+  }
+
+  return new S3Client(options);
+}
+
+/**
+ * Create CloudWatchLogs client with appropriate credentials
+ * 
+ * @param {string} role 
+ * @param {string} region 
+ * @returns {CloudWatchLogsClient}
+ */
+function createCloudWatchClient(role, region = 'us-east-1') {
+  const options = {
+    region: region,
+  }
+  if (role) {
+    let credentials;
+    try {
+      credentials = assumeRole(role);
+    } catch (error) {
+      throw new Error(`Unable to assume ${role}`, { cause: error });
+    }
+
+    options.credentials = {
+      // AWS CLI command will return credentials with 1st letter upper-case.
+      // However, S3 client expects different capitalization
+      ...Object.keys(credentials).reduce((memo, key) => {
+        memo[key.charAt(0).toLowerCase() + key.slice(1)] = credentials[key];
+        return memo;
+      }, {}),
+    }
+  }
+
+ return new CloudWatchLogsClient(options);
+}
+
+/**
+ * Get log streams for `options.logGroupName` (sort by last event time, limit 10)
+ * 
+ * @param {object} options 
+ * @param {string} options.logGroupName
+ * @param {string} options.role
+ * @param {string} options.region
+ * @param {CloudWatchLogsClient} client 
+ * @returns 
+ */
+async function getLogStreams(options, client) {
+  if (!client) {
+    client = createCloudWatchClient(options.role, options.region);
+  }
+
+  return await client.send(new DescribeLogStreamsCommand({
+    descending: true,
+    logGroupName: options.logGroupName,
+    orderBy: 'LastEventTime',
+    limit: 10,
+  }));
+}
+
+/**
+ * Get log events for `options.logStreamName`
+ * 
+ * @param {object} options 
+ * @param {string} options.logGroupName
+ * @param {string} options.logStreamName
+ * @param {string} options.role
+ * @param {string} options.region
+ * @param {CloudWatchLogsClient} client 
+ * @returns 
+ */
+async function getLogEvents(options, client) {
+  if (!client) {
+    client = createCloudWatchClient(options.role, options.region);
+  }
+
+  return await client.send(new GetLogEventsCommand({
+    logGroupName: options.logGroupName,
+    logStreamName: options.logStreamName,
+  }));
+}
+
+
+/**
+ * Only for testing: List all available buckets
+ * Ref: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/classes/listbucketscommand.html
+ * Req: options.role requires "s3:ListAllMyBuckets" permissions
+ * @param {object} options 
+ */
+async function listBuckets(options) {
```

Comment:
Just as double-check, *ListAllMyBuckets* perm is enabled in bulk use case? Because if not, I'd drop this method.

Reply:
It must be, as I've been using these methods as manual tests against your Psoxy instances...

Reply:
we shouldn't count on it. some customers deploy into shared accounts (even though we recommend against it)